packages/odyssey-react-mui/src/test-selectors/querySelector.ts

```diff
@@ -11,188 +11,238 @@
+          : FeatureTestSelector,
+      )(
+        // @ts-expect-error: Type '{ role?: AriaRole | undefined; options?: Record<LocalFeatureTestSelector extends FeatureSelector ? LocalFeatureTestSelector["feature"][FeatureName] extends TestSelector ? keyof LocalFeatureTestSelector["feature"][FeatureName]["selector"]["options"] : string : string, string | RegExp> | undefined; element: HTMLElement...' is not assignable to type '(LocalFeatureTestSelector extends FeatureSelector ? LocalFeatureTestSelector["feature"][FeatureName] : FeatureTestSelector) extends { ...; } ? Role extends AriaRole[] ? { ...; } : object : object'.ts(2345)
+        // `as featureTestSelector.feature[featureName]` narrows the props down enough that TypeScript errors here. We're passing the correct information, but it doesn't know that, and it's difficult to fix this. -Kevin Ghadyani
+        {
+          element: capturedElement,
+          queryMethod: childProps.queryMethod,
+          ...("options" in childProps && childProps.options
             ? {
-                role: templateArgsProp
-                  ? (interpolateString(
-                      testSelectors.selector?.role,
-                      templateArgsProp,
-                    ) as string)
-                  : testSelectors.selector?.role,
+                options: childProps.options as Options,
               }
-            : {
-                text: templateArgsProp
-                  ? interpolateString(
-                      testSelectors.selector?.text,
-                      templateArgsProp,
-                    )
-                  : testSelectors.selector?.text,
-              }),
-        })
-      : null;
-
-  const select =
-    "feature" in testSelectors
-      ? <FeatureName extends keyof (typeof testSelectors)["feature"]>(
-          featureName: FeatureName,
-          templateArgs?: (typeof testSelectors)["feature"][FeatureName] extends TestSelector
-            ? Record<
-                (typeof testSelectors)["feature"][FeatureName]["selector"]["templateVariableNames"][number],
-                string | RegExp
-              >
-            : never,
-        ) =>
-          querySelector({
-            canvas: element ? within(element) : canvas,
-            templateArgs,
-            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-            // @ts-expect-error: Type 'FeatureName' cannot be used to index type 'Record<string, FeatureTestSelector>'.ts(2536)
-            testSelectors: testSelectors.feature[featureName],
-          })
-      : null;
-
-  return {
-    element,
-    select,
+            : {}),
+          ...("role" in childProps && childProps.role
+            ? {
+                role: childProps.role as AriaRole,
+              }
+            : {}),
+        },
+      );
+    };
+
+    return {
+      element: capturedElement as LocalQueryMethod extends "get"
+        ? HTMLElement
+        : HTMLElement | null,
```

Comment:
get here doesn't have a nullable case, because it just throws in the situation where it can't find the element right? meanwhile find/query will just return a null possibly. For the case of query, would it ever return multiple elements, or is it always a single one like this?

Reply:
Only single. `queryAll` does multiple, but that's currently not supported.